analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.TryCatchFinally.cs

```diff
@@ -107,16 +119,66 @@ public void Finally_Nested_InstructionAfterFinally()
     Tag(""InOuterFinally"");
 }
 Tag(""AfterOuterFinally"");";
-            SETestContext.CreateCS(code).Validator.ValidateTagOrder(
+            var validator = SETestContext.CreateCS(code).Validator;
+            validator.ValidateTagOrder(
                 "BeforeOuterTry",
                 "InOuterTry",
                 "InInnerTry",
                 "InOuterFinally",       // With Exception thrown by Tag("InOuterTry")
                 "InInnerFinally",       // With Exception thrown by Tag("InInnerTry")
                 "InInnerFinally",
-                "AfterInnerFinally",
+                "AfterInnerFinally",    // Only once, because exception run from Tag("InInnerTry") continues directlyto outer finally
+                "InOuterFinally",
+                "AfterOuterFinally");
+
+            validator.TagStates("InInnerFinally").Should().HaveCount(2)
+                .And.ContainSingle(x => x.Exception == null)
+                .And.ContainSingle(x => x.Exception != null);
+            validator.TagStates("AfterInnerFinally").Should().HaveCount(1)  // Not visited by flows with Exception
+                .And.ContainSingle(x => x.Exception == null);
+            validator.TagStates("InOuterFinally").Should().HaveCount(2)
+                .And.ContainSingle(x => x.Exception == null)
+                .And.ContainSingle(x => x.Exception != null);
+            validator.TagStates("AfterOuterFinally").Should().HaveCount(1)  // Not visited by flows with Exception
+                .And.ContainSingle(x => x.Exception == null);
+            validator.ValidateExitReachCount(2);
+        }
+
+        [TestMethod]
+        public void Finally_Nested_InstructionAfterFinally_NoThrowsInFinally()
```

Comment:
Could you please add a test with try-finally in a finally block?

Reply:
Good catch, there was an issue with that scenario